packages/block-library/src/navigation-link/link-ui.js

```diff
@@ -193,47 +166,68 @@ export function LinkUI( props ) {
 			anchor={ props.anchor }
 			shift
 		>
-			<LinkControl
-				hasTextControl
-				hasRichPreviews
-				value={ link }
-				showInitialSuggestions={ true }
-				withCreateSuggestion={ userCanCreate }
-				createSuggestion={ handleCreate }
-				createSuggestionButtonText={ ( searchTerm ) => {
-					let format;
-
-					if ( type === 'post' ) {
-						/* translators: %s: search term. */
-						format = __( 'Create draft post: <mark>%s</mark>' );
-					} else {
-						/* translators: %s: search term. */
-						format = __( 'Create draft page: <mark>%s</mark>' );
+			{ ! addingBlock && (
+				<LinkControl
+					hasTextControl
+					hasRichPreviews
+					value={ link }
+					showInitialSuggestions={ true }
+					withCreateSuggestion={ userCanCreate }
+					createSuggestion={ handleCreate }
+					createSuggestionButtonText={ ( searchTerm ) => {
+						let format;
+
+						if ( type === 'post' ) {
+							/* translators: %s: search term. */
+							format = __( 'Create draft post: <mark>%s</mark>' );
+						} else {
+							/* translators: %s: search term. */
+							format = __( 'Create draft page: <mark>%s</mark>' );
+						}
+
+						return createInterpolateElement(
+							sprintf( format, searchTerm ),
+							{
+								mark: <mark />,
+							}
+						);
+					} }
+					noDirectEntry={ !! type }
+					noURLSuggestion={ !! type }
+					suggestionsQuery={ getSuggestionsQuery( type, kind ) }
+					onChange={ props.onChange }
+					onRemove={ props.onRemove }
+					onCancel={ props.onCancel }
+					renderControlBottom={ () =>
+						! link?.url?.length && (
+							<LinkUITools setAddingBlock={ setAddingBlock } />
+						)
 					}
+				/>
+			) }
+
+			{ addingBlock && (
+				<LinkUIBlockInserter
+					clientId={ props.clientId }
+					onBack={ setAddingBlock }
+				/>
+			) }
+		</Popover>
+	);
+}
 
-					return createInterpolateElement(
-						sprintf( format, searchTerm ),
-						{
-							mark: <mark />,
-						}
-					);
+function LinkUITools( { setAddingBlock } ) {
+	return (
+		<VStack className="link-ui-tools">
+			<Button
+				icon={ plus }
+				onClick={ ( e ) => {
+					e.preventDefault();
+					setAddingBlock( true );
 				} }
-				noDirectEntry={ !! type }
-				noURLSuggestion={ !! type }
-				suggestionsQuery={ getSuggestionsQuery( type, kind ) }
-				onChange={ props.onChange }
-				onRemove={ props.onRemove }
-				onCancel={ props.onCancel }
-				renderControlBottom={
-					! url
-						? () => (
-								<LinkControlTransforms
-									clientId={ props.clientId }
-								/>
-						  )
-						: null
-				}
-			/>
-		</Popover>
+			>
+				{ __( 'Add block' ) }
```

Comment:
Using voice over this announces Add block and the user ends up in a combo list of choices none of which is a block ...

Reply:
> ...none of which is a block ...

...they are all blocks really though right? 

What do you suggest to improve this?

Reply:
Maybe something more generic

```suggestion
				{ __( 'Add navigation item' ) }
```